templates/shared_services/firewall/terraform/variables.tf

```diff
@@ -1,6 +1,15 @@
-variable "tre_id" {}
+variable "tre_id" {
```

Comment:
think if we're passing in the `tre_id` we should be able to get the RG via a data block and use that location, to avoid having to pass the location as well?

Reply:
Either way works. I added the location back as it was before my PR  https://github.com/microsoft/AzureTRE/pull/1531/files#diff-244ec695609d33e2b1de334c5e0591a000b4ca09b6552d15559e22e5713b3957 for consitency with two other shared services. 

cc @tamirkamara  as he asked the same question

Reply:
I think to align on the approach regarding the location. My initial comment was that getting the RP to provide location is something we need to think carefully about, but as I also noted in my comment from yesterday, you have the core RG name anyway so you can use the RG data object which is what we use in all bundles (except workspace ones)